googlesource/acceptance_test.go

```diff
@@ -0,0 +1,273 @@
+}
+func (d AcceptanceTestDriver) Connector() sdk.Connector {
+	return d.Config.Connector
+}
+
+func (d AcceptanceTestDriver) SourceConfig(*testing.T) map[string]string {
+	return d.Config.SourceConfig
+}
+
+func (d AcceptanceTestDriver) BeforeTest(t *testing.T) {
+	// before test check if the test should be skipped
+	d.Skip(t)
+
+	if d.Config.BeforeTest != nil {
+		d.Config.BeforeTest(t)
+	}
+}
+
+func (d AcceptanceTestDriver) AfterTest(t *testing.T) {
+	if d.Config.AfterTest != nil {
+		d.Config.AfterTest(t)
+	}
+}
+
+func (d AcceptanceTestDriver) Skip(t *testing.T) {
+	var skipRegexs []*regexp.Regexp
+	for _, skipRegex := range d.Config.Skip {
+		r := regexp.MustCompile(skipRegex)
+		skipRegexs = append(skipRegexs, r)
+	}
+
+	for _, skipRegex := range skipRegexs {
+		if skipRegex.MatchString(t.Name()) {
+			t.Skip(fmt.Sprintf("caller requested to skip tests that match the regex %q", skipRegex.String()))
+		}
+	}
+}
+
+func (d AcceptanceTestDriver) GoleakOptions(_ *testing.T) []goleak.Option {
+	return d.Config.GoleakOptions
+}
+
+// WriteToSource writes data for source to pull data from
+func (d AcceptanceTestDriver) WriteToSource(t *testing.T, records []sdk.Record) []sdk.Record {
+	var err error
+	is := is.New(t)
+	config := d.SourceConfig(t)
+	records, err = writeToSource(t, config, records)
+	is.NoErr(err)
+
+	return records
+}
+
+func writeToSource(t *testing.T, config map[string]string, records []sdk.Record) (result []sdk.Record, err error) {
+	result, err = dataSetupWithRecord(t, config, records)
+	return result, err
+}
+func (d AcceptanceTestDriver) ReadFromDestination(*testing.T, []sdk.Record) []sdk.Record {
+	return []sdk.Record{}
+}
+
+func (d AcceptanceTestDriver) GenerateRecord(_ *testing.T) sdk.Record {
+	// we don't create record over here. Because we need to code here and then decode again
+	return sdk.Record{}
+}
+
+func createDatasetForAcceptance(t *testing.T, tableID string) (err error) {
+	ctx := context.Background()
+
+	client, err := bigquery.NewClient(ctx, projectID, option.WithCredentialsJSON([]byte(serviceAccount)))
+	if err != nil {
+		return fmt.Errorf("bigquery.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	meta := &bigquery.DatasetMetadata{
+		Location: location, // See https://cloud.google.com/bigquery/docs/locations
+	}
+
+	// create dataset
+	if err := client.Dataset(datasetID).Create(ctx, meta); err != nil && !strings.Contains(err.Error(), "duplicate") {
+		return err
+	}
+	t.Log("Dataset created")
+	client, err = bigquery.NewClient(ctx, projectID, option.WithCredentialsJSON([]byte(serviceAccount)))
+	if err != nil {
+		return fmt.Errorf("bigquery.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	sampleSchema := bigquery.Schema{
+		{Name: "abb", Type: bigquery.StringFieldType},
+		{Name: "name", Type: bigquery.StringFieldType},
+	}
+
+	metaData := &bigquery.TableMetadata{
+		Schema:         sampleSchema,
+		ExpirationTime: time.Now().AddDate(1, 0, 0), // Table will be automatically deleted in 1 year.
+	}
```

Comment:
Wouldn't an expiration time of a year be considered a bit too long for a test dataset? Also, we should probably remove the test data once the test is done.